python/idsse_common/test/test_publish_confirm.py

```diff
@@ -216,8 +209,37 @@ def test_publish_message_failure_rmq_error(publish_confirm: PublishConfirm, cont
 
 def test_on_channel_closed(publish_confirm: PublishConfirm, context: MockPika):
     publish_confirm.start()
-    sleep(.1)
-
     publish_confirm._on_channel_closed(context.Channel(), 'ChannelClosedByClient')
     assert publish_confirm._channel is None
     assert publish_confirm._connection.is_closed
+
+
+def test_start_with_callback(publish_confirm: PublishConfirm):
+    example_message = {'callback_executed': True}
+
+    def test_callback():
+        assert publish_confirm._channel.is_open
+        success = publish_confirm.publish_message(message=example_message)
+        assert success
+
+    assert publish_confirm._channel is None
+    publish_confirm.start(callback=test_callback)
+
+    sleep(.1)  # ensure that callback has time to run and send its message
+    assert publish_confirm._records.message_number == 1
+    assert publish_confirm._records.deliveries[1] == example_message
+
+
+def test_start_without_callback_sleeps(publish_confirm: PublishConfirm, monkeypatch: MonkeyPatch):
```

Comment:
I don't have any ideas for improvements, but this doesn't look like a great test. As I understand it, the mock sleep is possibly called by code other than PublishConfirm.start(), so I don't see how you know that 'mock_sleep.assert_called()' is testing that start() actually called sleep, just that mock_sleep is called at least once.

Reply:
Yeah, I went back and forth about this test. 

I originally asserted that sleep was called just once (the sleep call I'm targeting inside `PublishConfirm.start()`), but then it only passed some of the time--probably a race condition between my unit test and internal Thread sleeps. The mock said it was either called once, or called 10000 times.

I'll revert to my previous attempt to see if it's more consistent

Reply:
Ok, I updated the test now to manually count sleep calls only if it's for 0.2 seconds. It's ugly, and I don't like business logic being influenced by its testability, but that's the best way I found to uniquely pick out the sleep() call from `PublishConfirm.start()` vs others. 

Running in debug mode, I saw my mock catching `sleep(5)` (from `PublishConfirm.run()`) and `sleep(0.1)` (probably from python Thread or unit test operations)

Reply:
I'm with you, I don't know what way to go. 
The previous test could pass if PublishConfirm.start() is never called.
This test could fail if something else calls sleep(0.2) in addition to PublishConfirm.start().
I think I'd go with too lenient and let it pass even if maybe it shouldn't. So I'd use something like assert.called_with(.2), which check that sleep is called with the same length of time that PublishConfirm.start() would call it with, but would still pass if something else also calls sleep(0.2). 
Also is there a reason you didn't use assert.called_once_with(.2) vs the custom counter in the current code?
I don't want to hold this PR up anymore so do what you think is best and merge.

Reply:
`assert_called_once_with()` just had a race condition where the unit test would sometimes pass, but sometimes fail with 
```
 AssertionError: Expected 'mock' to be called once. Called 2 times.
E           Calls: [call(5), call(0.2)].
```

My latest code asserts that sleep was _at least_ called with 0.2 seconds, but doesn't fail if additional sleep calls were captured.